devito/types/array.py

```diff
@@ -368,6 +368,8 @@ def __args_setup__(cls, *args, **kwargs):
             raise ValueError("Component type must be subclass of AbstractFunction")
         if len({i.__padding_dtype__ for i in components}) != 1:
             raise ValueError("Components must have the same padding dtype")
+        if len({i.properties for i in components}) != 1:
+            raise ValueError("Components must have the same properties")
```

Comment:
looks more like "counting" rather than "comparing", should you  say, the same number of properties?

Reply:
It's not counting properties it's check the set length one, i.e all properties are the same